news/service/db.py

```diff
@@ -0,0 +1,18 @@
+from pymongo import MongoClient
+
+_DB_CLIENT = MongoClient("mongodb://news-db:27017/news-db")
+_DB_CLUSTERED_ARTICLES = _DB_CLIENT['news-db']['clusteredarticles']
+
+
+def fetch_n_clusters_before_time(timestamp: int, n_clusters: int):
+    return _DB_CLUSTERED_ARTICLES \
+        .find({'timestamp': {'$lt': timestamp}}) \
+        .sort([('timestamp', -1)]) \
```

Comment:
why does sort direction change?

Reply:
to be able to locate `next_timestamp` in resulting query, as explained above

Reply:
actually my last comment is wrong, I did it because I want to find the X clusters happening before timestamp. `.limit` is going to take the first X encountered in the query result, hence the reversed sorting

Reply:
Ah that makes sense. You should sort the results in python then so that timestamp is always increasing